### PR TITLE
3813 likes counts doesn't match

### LINKED ISSUE #3813: Likes counts doesn't match
In my local environment likes counts don't match -more information to come-.

### PR DESCRIPTION
**Pull request by juanignaciosl**
This fixes #3813 by adding a couple of checks supporting inconsistent data. Not something I'd like, but who knows when data will be 100% consistent. @Kartones, CR this, please.

As you see this PR comes with a gift: I've moved UUIDHelper for reload to work again ;-). Buuuuut... path at `require_depenency` is strange. A relative path didn't work in local, and using `lib/carto/uuidhelper` didn't work in staging. I guess `/` is not configured as search path in staging and lib is, since `carto/uuidhelper` works both in local and staging. @zenitraM , could you confirm this path will work in production?

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3957/))

File: app/controllers/carto/api/layers_controller.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 
-require_relative '../../../helpers/carto/uuidhelper.rb'
+require_dependency 'carto/uuidhelper'
```
Comment (Kartones) [inline app/controllers/carto/api/layers_controller.rb]: <3

File: app/models/visualization/stats.rb [hunk 2]
```diff
@@ -16,7 +16,7 @@ def initialize(visualization, user=nil)
       
       def to_poro
         new_calls = {}
-        CartoDB::Stats::APICalls.new.get_api_calls_with_dates(user.username, {stat_tag: visualization.id}).to_a.reverse.each do |call|
+        CartoDB::Stats::APICalls.new.get_api_calls_with_dates(username, {stat_tag: visualization.id}).to_a.reverse.each do |call|
```
Comment (Kartones) [inline app/models/visualization/stats.rb]: does this `get_api_calls_with_dates` support empty username and returns no results?

**Comment by Kartones** [comment]
:+1:

**Comment by zenitraM** [comment]
There's next to none difference between staging environments and prod, in ruby/conf versions... so whatever happens in Staging should be on Production..

could it be due to Ruby 2?

File: app/models/visualization/stats.rb (hunk 2 above)
Comment (juanignaciosl) [inline app/models/visualization/stats.rb]: It ends in a Redis query with empty username, so no result will be return.

**Comment by juanignaciosl** [comment]
@zenitraM I checked and it also worked in local. Development config or defaults, I guess. I'll trust staging then.

File: app/models/visualization/stats.rb (hunk 2 above)
Comment (Kartones) [inline app/models/visualization/stats.rb]: then maybe would be better to not call no? and avoid querying redis for nothing

Comment (juanignaciosl) [inline app/models/visualization/stats.rb]: Ok... I  didn't want to add extra logic to support not consistent data, but you're right :disappointed:

**Pull request merged**